extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -14,5 +14,73 @@
  * limitations under the License.
  */
 
-export class AmpViewerIntegration {
+import {Messaging} from './messaging.js';
+import {listen, unlisten} from '../../../src/event-helper';
+import {viewerPromiseForDoc} from '../../../src/viewer';
+import {user} from '../../../src/log';
+
+
+class AmpViewerIntegration {
+
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @const @private {!Window} win */
+    this.win_ = win;
+  }
+
+  init() {
+    let viewer;
+    this.getViewer()
+    .then(viewerParam => {
+      viewer = viewerParam;
+      return this.getHandshakePromise(viewer);
+    })
+    .then(viewerOrigin => {
+      const messaging = new Messaging(this.win_.parent, viewerOrigin,
+          function(type, payload, awaitResponse) {
+            return viewer.receiveMessage(type, payload, awaitResponse);
```

Comment:
Ok. Can let's keep this one as is: we'd need to refactor it a bit deeper. We can do it later.